kafka/consumer.go

```diff
@@ -95,15 +86,14 @@ func (cfg *ConsumerConfig) finalize() error {
 	if cfg.GroupID == "" {
 		errs = append(errs, errors.New("kafka: consumer GroupID must be set"))
 	}
-	if cfg.Decoder == nil {
-		errs = append(errs, errors.New("kafka: decoder must be set"))
-	}
 	if cfg.Processor == nil {
 		errs = append(errs, errors.New("kafka: processor must be set"))
 	}
 	return errors.Join(errs...)
 }
 
+var _ apmqueue.Consumer = &Consumer{}
```

Comment:
nit (feel free to ignore): I assume this it to check if `apmqueue.Consumer` is implemented by *Consumer. Maybe we can do something like below:

```suggestion
var _ apmqueue.Consumer = (*Consumer)(nil)
```